demisto_sdk/commands/format/update_generic_yml.py

```diff
@@ -143,7 +147,29 @@ def update_tests(self) -> None:
                             test_playbook_data = get_yaml(tpb_file_path)
                             test_playbook_id = get_entity_id_by_entity_type(test_playbook_data,
                                                                             content_entity='')
-                            test_playbook_ids.append(test_playbook_id)
+                            if not scripts and not commands:  # Better safe than sorry
+                                test_playbook_ids.append(test_playbook_id)
+                            else:
+                                added = False
+                                tpb_commands, tpb_scripts = get_scripts_and_commands_from_yml_data(
+                                    test_playbook_data, FileType.TEST_PLAYBOOK)
+
+                                for full_tpb_command in tpb_commands:
+                                    tpb_command_parts = full_tpb_command.split('|||')
+                                    tpb_command = full_tpb_command
+                                    if len(tpb_command_parts) == 2:
+                                        if file_id and file_id != tpb_command_parts[0]:
+                                            continue
+                                        tpb_command = tpb_command_parts[1]
```

Comment:
As said in the tools, perhaps better to move that logic inside the tools func. WDYT?

Reply:
I changed things up a little, I think it looks better now, I didn't fully understand what you suggested, but let me know what you think of this solustion